config/default/manager_webhook_patch.yaml

```diff
@@ -1,15 +1,22 @@
-apiVersion: apps/v1
-kind: Deployment
-metadata:
-  name: operator
-  namespace: vm
-spec:
-  template:
-    spec:
-      containers:
-      - name: manager
-        ports:
-        - containerPort: 9443
-          name: webhook-server
-          protocol: TCP
-
+- op: add
+  path: /spec/template/spec/containers/0/args/-
+  value: --webhook.enable
+- op: add
+  path: /spec/template/spec/containers/0/ports/-
+  value:
+    containerPort: 9443
+    name: webhook-server
+    protocol: TCP
+- op: add
+  path: /spec/template/spec/containers/0/volumeMounts/-
+  value:
+    mountPath: /tmp/k8s-webhook-server/serving-certs
+    name: cert
+    readOnly: true
+- op: add
```

Comment:
it ll produce incorrect deployment definition for olm package.

```yaml
                    volumeMounts:
                      - mountPath: /tmp/k8s-webhook-server/serving-certs
                        name: cert
                        readOnly: true
                securityContext:
                  runAsNonRoot: true
                  seccompProfile:
                    type: RuntimeDefault
                serviceAccountName: vm-operator
                terminationGracePeriodSeconds: 10
                volumes:
                  - name: cert
                    secret:
                      defaultMode: 420
                      secretName: webhook-serving-cert
```
I suggest to keep focus on OLM compatibility for kustomize manifests.

 First of all, it must produce valid, reproducible builds for operatorhub.

It's not recommend way for installing operator.

Reply:
you can check other operators in redhat repo, they also have secretName mounted

Reply:
For readhat it could be fine, since openshift install each operator at own namespace.
 
 But for operatorhub it seems to be not, since it's not possible to share secretName: webhook-serving-cert with other operators at `operators` namespace. Only if it ll be patched with some magic.

Reply:
added a separate overlay `base` for kustomize, `default` is a shared one for `olm` and `base`